Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/search/SearchPage.tsx

```diff
@@ -19,43 +17,59 @@ import styles from './SearchPage.module.scss';
 const SearchPage = () => {
     const {
         state: {
-            contests,
-            problems,
-            users,
             searchError,
             searchValue,
-            isLoaded,
+            getSearchResultsUrlParams,
         },
-        actions: { initiateSearchResultsUrlQuery },
-    } = useSearch();
-    const {
-        state: {
-            currentPage,
-            pagesInfo,
+        actions: {
+            initiateSearchResultsUrlQuery,
+            setSearchingError,
+            toggleVisibility,
         },
-        changePage,
-    } = usePages();
+    } = useSearch();
     const { actions: { setPageTitle } } = usePageTitles();
+    const [ isSearchingContests, setIsSearchingContests ] = useState(false);
```

Comment:
Can you also transfer those states inside the hook? Keep the components as clean as possible.

Reply:
State variables and logic are transferred to the hook.